### ISSUE TITLE
sync: upload error is not retried nor does it exit with error

### ISSUE THREAD

**Issue by alanfranz** [Backblaze/B2_Command_Line_Tool#163]
b2 client 0.5.4. (updated to 0.5.6 when editing comment and reporting point 3)

Today I got a lot of connection reset errors while performing a sync command:

```
b2_upload(/other/localbackup/duplicity-full.20160526T124808Z.vol55.difftar.gpg, duplicity-full.20160526T124808Z.vol55.difftar.gpg, 1464271210848): UnknownError() unknown error: SysCallError(104, 'ECONNRESET')
```

On the webapp, I see a "started large file" with 0 bytes and that name.

I don't know why, I suppose some transient network issues from my network to backblaze servers, but the real issue is:
1) The sync command does not retry to upload the file, ever.
2) The sync command does not exit with an error status in such situation! Since my purpose is to employ the b2 client as an unattended uploader for backups, that's quite an issue!
3) launching again the sync command, it will do nothing; it won't try to resync the wrongly uploaded files - I must force the --compareVersions size option!

IMHO:
1) the sync command should retry a certain number of times the upload if the error is transient (or  maybe, if it's hard to understand wheter it's transient, it could retry a certain number of times and reset the counter if at any time succeeds);
2) if ANY unrecovered error happens during the sync, the exit status from the tool should be != 0; I think this is really critical in production environments.
3) If some error happens, a removal of the destination large file should be attempted, to prevent failed uploads to be corrected by further syncs (or any other measure that can prevent such behaviour would be ok).

**Comment by svonohr** [comment]
Thanks for reporting this bug. Regarding your comments:
1) This should already happen for some errors. I have to check if it will also happen for your error.
2) This also should be the case, when looking at the code. The exit status should be 1 in case of an error but I have to verify this.
3) I've just created a branch where this is fixed. Started large files are now ignored by the sync command, so launching the sync command again sees this file as missing and will upload it. It will even resume the previous upload if the file didn't change.

**Comment by alanfranz** [comment]
@svonohr would you like me to split this issue into separate issues for 1) and 2), with a specific test to reproduce?

**Comment by svonohr** [comment]
Yes, splitting the issue would be useful. This seems to be 3 separate bugs.

**Comment by svonohr** [comment]
Issue 3 was already fixed in #166. I have created new issues for the other two problems.
